tests/e2e/tests/pilot/util/environment.go

```diff
@@ -869,3 +966,51 @@ func (e *Environment) Fill(inFile string, values interface{}) (string, error) {
 
 	return out.String(), nil
 }
+
+func (e *Environment) createMulticlusterConfig() error {
+	// Although this function loops through all files in the configuration directory the tests assumes a single
+	// clusterregistry configuration file.
+
+	info, err := os.Stat(e.Config.ClusterRegistriesDir)
+	if err != nil {
+		switch err := err.(type) {
+		case *os.PathError:
+			return fmt.Errorf("error reading %s: %v", e.Config.ClusterRegistriesDir, err.Err)
+		default:
+			return fmt.Errorf("error reading %s: %v", e.Config.ClusterRegistriesDir, err)
+		}
+	}
+
+	if info.IsDir() {
+		if strings.Contains(e.Config.ClusterRegistriesDir, "=") {
+			return fmt.Errorf("cannot give a key name for a directory path")
+		}
+		fileList, err := ioutil.ReadDir(e.Config.ClusterRegistriesDir)
+		if err != nil {
+			return fmt.Errorf("error listing files in %s: %v", e.Config.ClusterRegistriesDir, err)
+		}
+		var Data []byte
+		configData := make(map[string]string)
+		for _, item := range fileList {
+			itemPath := path.Join(e.Config.ClusterRegistriesDir, item.Name())
+			if item.Mode().IsRegular() {
+				keyName := item.Name()
+				Data, err = ioutil.ReadFile(itemPath)
+				configData[keyName] = string(Data)
```

Comment:
Did you want to assign an empty string to the configData, even when there is an error reading the file? IOW, should this be after the error check?

Reply:
done